upstash/qstash/schedule/api_calls.go

```diff
@@ -25,25 +25,29 @@ func getSchedule(c *client.UpstashClient, scheduleId string) (schedule QstashSch
 	return schedule, err
 }
 
-func createSchedule(c *client.UpstashClient, body CreateQstashScheduleRequest) (schedule QstashSchedule, err error) {
+func deleteSchedule(c *client.UpstashClient, scheduleId string) (err error) {
+	return c.SendDeleteRequest(c.GetQstashEndpoint()+"/schedules/"+scheduleId, nil, "Delete QStash Schedule", true)
+}
+
+func createSchedule(c *client.UpstashClient, body CreateQstashScheduleRequest) (scheduleID string, err error) {
 
 	err, BEARER_TOKEN := c.GetQstashToken()
 	if err != nil {
-		return schedule, err
+		return "", err
 	}
-	endpoint := c.GetQstashEndpoint() + "/publish/" + body.Destination
+	endpoint := c.GetQstashEndpoint() + "/schedules/" + body.Destination
 
 	postParameters := []interface{}{
-		req.Header{"Upstash-Cron": body.Headers.Cron},
-		req.Header{"Upstash-Retries": fmt.Sprint(body.Headers.Retries)},
-		req.Header{"Accept": "application/json"},
-		req.Header{"Authorization": "Bearer " + BEARER_TOKEN},
 		req.Header{"Content-Type": body.Headers.ContentType},
-		req.Header{"Upstash-Deduplication-Id": body.Headers.DeduplicationId},
-		req.Header{"Upstash-Content-Based-Deduplication": fmt.Sprint(body.Headers.ContentBasedDeduplication)},
-		req.Header{"Upstash-NotBefore": fmt.Sprint(body.Headers.NotBefore)},
+		req.Header{"Upstash-Method": body.Headers.Method},
 		req.Header{"Upstash-Delay": body.Headers.Delay},
+		req.Header{"Upstash-Retries": fmt.Sprint(body.Headers.Retries)},
```

Comment:
These header implementation would be in dynamic with `headerMap := map[string]string{}` and we can iterate it